src/AsyncEventDispatcher.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ArchiPro\EventDispatcher;
+
+use function Amp\async;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\EventDispatcher\ListenerProviderInterface;
+use Psr\EventDispatcher\StoppableEventInterface;
+
+use Revolt\EventLoop;
+
+/**
+ * Asynchronous implementation of PSR-14 EventDispatcherInterface using Revolt and AMPHP.
+ *
+ * This dispatcher schedules event listeners to be executed asynchronously using the Revolt event loop.
+ * The dispatch method returns immediately without waiting for listeners to complete.
+ */
+class AsyncEventDispatcher implements EventDispatcherInterface
+{
+    private ListenerProviderInterface $listenerProvider;
+
+    /**
+     * @param ListenerProviderInterface $listenerProvider The provider of event listeners
+     */
+    public function __construct(
+        ListenerProviderInterface $listenerProvider
+    ) {
+        $this->listenerProvider = $listenerProvider;
+    }
+
+    /**
+     * Dispatches an event to all registered listeners asynchronously.
+     *
+     * Each listener is scheduled in the event loop and executed asynchronously.
+     * The method returns immediately without waiting for listeners to complete.
+     * If the event implements StoppableEventInterface, propagation can be stopped
+     * to prevent subsequent listeners from being scheduled.
+     *
+     * @param object $event The event to dispatch
```

Comment:
`AbstractEvent` implements `StoppableEventInterface`, is this intentional? If it is, what's the base interface for all events to use in that case?

Following up on that, the `$event` arg should be typed to that base interface/class for all events.

EDIT: I didn't read the comment on `AbstractEvent`. I think it's better to rename it to be explicit, `AbstractEvent` doesn't right away tell the user that this is the stoppable version. 

`$event` should be typed to whatever the PSR14 interface for an event is.